### DISCUSSION TITLE
Force https and use custom css

### DISCUSSION THREAD

**Discussion by Michelle-Luttringer** [GetPublii/Publii#1412] [Help]
Hy everyone,
I have two questions.
First, the ssl certificate got installed for my website and I swiched the server configuration to https. Now if I tipe in my URL on a Browser, it works with most browsers and my website gets shown as secure. But with some browsers it will go to the non secure http version and it only works if I forece https by tiping it in. Is there a way to automaticaly force https? I havent seen anything in the server configurations, but maby somebody knows if there is a way.

Second question, if I whant to change things in the theme, did I get that right, that I can do that via custom css. I saw there is a box called custom css with the plugins. Can I just write my changes in there or is there anything else I have to do, to make it work?

Thanks for your help.

**Comment by example-user30** [comment]
Your host should have an option to force the site to use https://.  In some cases you might have to force that with a setting in an .htaccess file - read about that [here](https://www.answerlookup.com/how-force-https-using-htaccess-apache-or-nginx).  On my host I had the option to force https:// from the hosting service control panel.

It is always best to post dissimilar questions in two different threads so other people can benefit from the answers in the future.  
That said - 

Yes - just put well formed CSS into the Custom CSS section of the Tools & Plugins section.  You should not put the style tags in there though.  As an example - I like to make the menubar on simple theme black with an opacity of .5.  I do this by putting the following into the Custom CSS:

```css:
@media (min-width: 56.25em) {
    .top {
        background-color: rgba(0,0,0,.5);
    }
}
```

**Comment by candidexmedia** [comment]
Re your CSS question: Custom CSS in the GUI works great.

That said, I find that the in-app code editor isn't ideal for major CSS changes and editing. As someone who heavily modifies Publii themes to create custom sites, I prefer going the [theme override route](https://getpublii.com/dev/theme-overrides/) and editing `main.css` directly using VS Codium.

This allows me to modify the stylesheet in its context (rather than relying on overrides) and understand how the styling works. It also allows me to leverage VS Code shortcuts and plugins to make the work more efficient.

When I edit `main.css`, I make sure to add comments `/* like this */` to track every new insertion and modification.

When a new theme version is released, I make sure to compare my theme override `main.css` against the new one to see how updates impact my existing modifications.

Another note: some themes manage colour variables and typography (family, weight, [fluid/responsive sizing](https://github.com/GetPublii/Publii/discussions/1384#discussioncomment-8599536)) using [theme variables](https://getpublii.com/dev/theme-variables/) and [visual overrides](https://getpublii.com/dev/visual-overrides/), so it may be best to edit those theme files directly, rather than using Custom CSS (see for example - [how to add a custom font](https://getpublii.com/docs/fonts.html#adding-custom-font)).

**Reply by example-user30** [reply]
I hear you.  I am not a theme developer (yet).  I find that doing simple overrides in the Custom CSS for small changes to an existing theme is an easy route to go since I am not a theme developer. 

Thanks
